lib/json_api_resource/resource.rb

```diff
@@ -54,7 +54,13 @@ def method_missing(method, *args, &block)
       if match = method.to_s.match(/^(.*)=$/)
         self.client.send(match[1], args.first)
       elsif self.client.respond_to?(method.to_sym)
-        self.client.send(method)
+        is_method = self.client.methods.include?(method.to_sym)
+        argument_count = is_method ? self.client.method(method.to_sym).arity : 0
+        if argument_count > 0 || argument_count == -1
```

Comment:
self.client.method(method.to_sym) will throw an exception if method.to_sym is not really a method.   So you have to check first if methods.include?(method.to_sym).  But otherwise it is rearrangable as you suggest.